connectors/sources/mongo.py

```diff
@@ -134,13 +134,20 @@ def get_default_configuration(cls):
                 "type": "str",
                 "value": "",
             },
-            "user": {"label": "Username", "order": 5, "type": "str", "value": ""},
+            "user": {
+                "label": "Username",
+                "order": 5,
+                "type": "str",
+                "value": "",
+                "required": False,
+            },
```

Comment:
If username and password are empty, MongoDB will just run without them, correct?
We could hide this behind a `depends_on` on a boolean field like `use_authentication` or something. Not required, but if you want that functionality we can have it.

Reply:
I think this is a good idea - with defaulting `use_authentication` to `true`.

Reply:
I hesitated to do so to allow seamless migration from Ruby to Python MongoDB connector, idk if it's a big deal.

@danajuratoni what do you think?

Reply:
I see username and password are not mandatory, so the functionality is already there. Since I don't expect prod env with no authentication, we're good to keep the UI changes at a minimum for now and avoid adding a new field.